openml/tasks/task.py

```diff
@@ -7,38 +7,58 @@
 import openml._api_calls
 from ..utils import _create_cache_directory_for_id
 
-
 class OpenMLTask(object):
-    def __init__(self, task_id, task_type_id, task_type, data_set_id,
-                 target_name, estimation_procedure_type, data_splits_url,
-                 estimation_parameters, evaluation_measure, cost_matrix,
-                 class_labels=None):
+    def __init__(self, task_id, task_type_id, task_type, data_set_id, estimation_procedure_type,
+                 estimation_parameters, evaluation_measure):
         self.task_id = int(task_id)
         self.task_type_id = int(task_type_id)
         self.task_type = task_type
         self.dataset_id = int(data_set_id)
-        self.target_name = target_name
         self.estimation_procedure = dict()
         self.estimation_procedure["type"] = estimation_procedure_type
-        self.estimation_procedure["data_splits_url"] = data_splits_url
         self.estimation_procedure["parameters"] = estimation_parameters
         #
         self.estimation_parameters = estimation_parameters
         self.evaluation_measure = evaluation_measure
-        self.cost_matrix = cost_matrix
-        self.class_labels = class_labels
-        self.split = None
-
-        if cost_matrix is not None:
-            raise NotImplementedError("Costmatrix")
 
     def get_dataset(self):
         """Download dataset associated with task"""
         return datasets.get_dataset(self.dataset_id)
 
+    def push_tag(self, tag):
```

Comment:
didn't we have general tag functions?

Reply:
Otherwise we should create them. (and not duplicate tag functions code for each entity)

Reply:
Great remark. I think we should do this in a new PR, though, as it is not directly related to this PR. I opened issue #549 for this.